docs/naps/0-nap-process.md

```diff
@@ -0,0 +1,297 @@
+way to do this.
+
+The proposal should be submitted as a draft NAP via a [GitHub pull
+request][github pull request] to the `docs/source/naps` directory with the
+name `nap-<n>-<short-title>.md` where `<n>` is an appropriately assigned
+number and `<short-title>` is a one or two word title for the idea (e.g.,
+`nap-35-lazy-slicing.md`). The draft must use the {ref}`nap_template` file.
+
+Once the PR is in place, the NAP should be announced on various channels
+for discussion, including the [#naps channel on Zulip]() and, if the NAP
+has significant user implications, on the [image.sc forum]().
+
+At the earliest convenience, the PR should be merged (regardless of whether
+it is accepted during discussion). A NAP that outlines a coherent argument
+and that is considered reasonably complete should be merged optimistically,
+regardless of whether it is accepted during discussion. Additional PRs may
+be made by the author to update or expand the NAP, or by maintainers to set
+its status, discussion URL, etc.
+
+Standards Track NAPs consist of two parts, a design document and a
+reference implementation. It is generally recommended that at least a
+prototype implementation be co-developed with the NAP, as ideas that sound
+good in principle sometimes turn out to be impractical. Often it makes
+sense for the prototype implementation to be made available as PR to the
+napari repo, as long as it is properly marked as WIP (work in progress).
+
+### Review and Resolution
+
+NAPs are discussed in Zulip, on image.sc, and on GitHub. The possible paths
+of the status of NAPs are as follows:
+
+```{image} _static/nap-flowchart.png
+```
+
+All NAPs should be created with the `Draft` status.
+
+The author of the NAP should periodically update the NAP with new points
+both against and in favor of the NAP raised in discussion.
+
+Eventually, after discussion, there may be a consensus that the NAP
+should be accepted – see the next section for details. At this point
+the status becomes `Accepted`.
+
+Once a NAP has been `Accepted`, the reference implementation must be
+completed. When the reference implementation is complete and incorporated
+into the main source code repository, the status will be changed to
+`Final`.
+
+To allow gathering of additional design and interface feedback before
+committing to long term stability for a language feature or standard
+library API, a NAP may also be marked as "Provisional". This is short for
+"Provisionally Accepted", and indicates that the proposal has been accepted
+for inclusion in the reference implementation, but additional user feedback
+is needed before the full design can be considered "Final". Unlike regular
+accepted NAPs, provisionally accepted NAPs may still be Rejected or
+Withdrawn even after the related changes have been included in a release.
+
+Wherever possible, it is considered preferable to reduce the scope of a
+proposal to avoid the need to rely on the "Provisional" status (e.g. by
```

Comment:
Yeah, agreed - this is also why I like the idea that something "Provisional" has to actually graduated into "Accepted", but maybe I'm not quite appreciating all the nuance here

Reply:
I don't think it's that you're not appreciating the nuance, but rather, you're overthinking the nuance. I think that in practice, you typically go Provisional -> (implement) -> Final or Provisional -> (implement) -> (oops, actually huge problems) -> (reject/withdraw).

If anything, I feel like the line between Accepted and Final is blurry. But in all of these cases, I don't think it's worth overthinking it vs adopting a process that has been in place for at least two decades and as far as I know not really caused any major furore (other than the walrus operator, and that seems to have died down, and anyway I don't think this nitpick would have saved the situation there :joy:)